PacketDotNet/WakeOnLanPacket.cs

```diff
@@ -148,6 +148,37 @@ public byte[] Password
                 }
                 return new byte[0];
             }
+            set
+            { 
+                var bytes = value;
+                var headerLength = WakeOnLanFields.PasswordPosition;
+                var newLength = headerLength + bytes.Length;
+                var prevValueLength = Password.Length;
+
+                //If Wake-On-Lan packet size is not matching, create a new ByteArraySegment.
+                if (bytes.Length != prevValueLength)
+                {
+
+                    // allocate new memory for this packet
+                    var newByte = new byte[newLength];
+
+                    // copy the header bytes over
+                    Array.Copy(Header.Bytes, Header.Offset,
+                               newByte, 0,
+                               headerLength);
+
+                    Header = new ByteArraySegment(newByte, 0, newLength);
```

Comment:
Reassigning Header is a dangerous operation as other content might depend on it or use the bytes assigned elsewhere. Generally speaking, this is something user must do on its own rather that the library does this (this usually involves a manual move of the existing bytes of the top most packet). Therefore I think this entire block of moving bytes around should be deleted.
An exception can be thrown if the given bytes length isn't 6 or 4.

Reply:
I understand there can be a issue here. But this implementation will provide better usability.
Note: Similar implementation is in LLDP OrganizationSpecific.cs.
And higher layer protocols has similar need for reassigning requirement. So i think we should consider this.
Please let me know your comment.

Reply:
I disagree, we should avoid setting Header (same in LLDP for that matter). Most of the time you're going to create a packet starting with a byte array in EthernetPacket. If you're going to override Header then the original byte array will no longer be modified and you're ending up with disjoint packets.

Reply:
Ok, then consider the case when Password was of length 6 bytes, and i want to change it to 4 bytes. This cannot be done, if header reassignment is not supported. So how should the setter handle it.

Reply:
The setter shouldn't, similar to all the other properties with variable length. If you're changing this then you should first move the original bytes to the right (or left if you're shrinking).

Reply:
Changes has been maded